ecosystem/typescript/sdk/README.md

```diff
@@ -31,39 +24,51 @@ and `allowSyntheticDefaultImports` in your `tsconfig` for types compatibility:
 ```
 
 ### Requirements
-
 - [Node.js](https://nodejs.org)
-- [yarn](https://yarnpkg.com/)
+- [Yarn](https://yarnpkg.com/)
 
 ```bash
-sudo apt-get update
-sudo apt-get install nodejs yarn
+yarn install
 ```
 
-### Generating Types
+### Generating API client
+This SDK is composed of two parts, a core client generated from the OpenAPI spec of the API, and a set of wrappers that make it nicer to use, enable certain content types, etc.
 
-Originally created with this:
+To generate the core client from the spec, run:
+```bash
+yarn generate-client
+```
 
+### Testing against devnet
 ```bash
-$  npx swagger-typescript-api@latest -p ../../../api/doc/v0/openapi.yaml -o ./src/api --modular --axios --single-http-client
+yarn test
 ```
 
-#### Changes to make after generation:
+### Testing against local node
 
-- OpenAPI/SpecHTML routes/types deleted as they're unneeded.
-- There are a few type errors in the `http-client.ts` as the axios types are incomplete, that were fixed
-  via `// @ts-ignore`
+Run a local node (run from the root of the repo):
+```
+cargo run -p aptos -- node run-local-testnet --with-faucet --faucet-port 8081 --force-restart --assume-yes
+```
 
-### Testing (jest)
+Setup an env to configure the URLs:
+```
+rm .env
+echo 'APTOS_NODE_URL="http://127.0.0.1:8080/v1"' >> .env
+echo 'APTOS_FAUCET_URL="http://127.0.01:8081"' >> .env
+```
 
-```bash
+Run the tests:
+```
 yarn test
 ```
```

Comment:
Why bother creating `.env`? Why not just `APTOS_NODE_URL="http://127.0.0.1:8080/v1 APTOS_FAUCET_URL="http://127.0.01:8081 yarn test`. No side effects.

Reply:
That command doesn't seem to work for me, it fails to connect to my locally running node. But the `.env` approach works. This seems to be a sort of common problem: https://stackoverflow.com/questions/50259025/using-env-files-for-unit-testing-with-jest.